### DISCUSSION TITLE
How does TF-Watcher work under the hood? The data (loss-values), how is it stored or send and to whom, so that the "app" can access and display the data?

### DISCUSSION THREAD

**Discussion by RomanFoell** [Rishit-dagli/TF-Watcher#98] [General]
Hello together,

the question in the title is clear, can you tell me more about that?
What is, when the Server on which the python code is running does not have internet-access, but can connect remotely to a laptop, which has internet-access? Does it still work then somehow?

Best, Roman

**Comment by Rishit-dagli** [comment]
Hi @RomanFoell ,
The metrics you choose to monitor are taken from when you call `model.compile` or override the `compile` function in subclassed model and are fetched through [TensorFlow Callbacks](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/) at the intervals you specify. These are then sent over to Firebase Realtime Database. I would encourage you to check out the implementation for [any of the classes](https://github.com/Rishit-dagli/TF-Watcher/blob/main/tfwatcher/callbacks/epoch.py).
